runtime/parser/parse_partial_data.go

```diff
@@ -10,13 +10,14 @@ import (
 // DataYAML is the raw YAML structure of a sub-property for defining a data resolver and properties.
 // It is used across multiple resources, usually under "data:", but inlined for APIs.
 type DataYAML struct {
-	Connector      string         `yaml:"connector"`
-	SQL            string         `yaml:"sql"`
-	MetricsSQL     string         `yaml:"metrics_sql"`
-	API            string         `yaml:"api"`
-	Args           map[string]any `yaml:"args"`
-	Glob           yaml.Node      `yaml:"glob"` // Path (string) or properties (map[string]any)
-	ResourceStatus map[string]any `yaml:"resource_status"`
+	Connector       string         `yaml:"connector"`
+	SQL             string         `yaml:"sql"`
+	MetricsSQL      string         `yaml:"metrics_sql"`
+	AdditionalWhere string         `yaml:"additional_where"` // Optional additional WHERE clause for metrics SQL
```

Comment:
Should not be an option in `DataYAML`, only in the `metrics_sql.go` resolver (since this is something that shouldn't be used from YAML, only in requests).